ts/components/wallet/WalletLayout.tsx

```diff
@@ -7,111 +7,111 @@
+    paddingHorizontal: customVariables.contentPadding
   },
 
-  noalias: {
-    marginRight: 0
+  whiteBg: {
+    backgroundColor: customVariables.colorWhite
   },
 
-  white: {
-    color: variables.colorWhite
+  flex1: {
+    flex: 1
   },
 
-  whiteBg: {
-    backgroundColor: variables.colorWhite
+  brandDarkGray: {
+    color: customVariables.brandDarkGray
   },
 
-  noBottomPadding: {
-    padding: variables.contentPadding,
-    paddingBottom: 0
+  subHeaderContent: {
+    flexDirection: "row",
+    alignItems: "baseline",
+    justifyContent: "space-between",
+    paddingHorizontal: customVariables.contentPadding,
+    backgroundColor: customVariables.colorWhite
   }
 });
 
-type Props = Readonly<{
-  title: string;
-  headerContents?: React.ReactNode;
-  onNewPaymentPress?: () => void;
-  allowGoBack: boolean;
-  displayedWallets?: React.ReactNode;
-  contentStyle?: StyleProp<ViewStyle>;
-}>;
-
 export default class WalletLayout extends React.Component<Props> {
-  private WalletLayoutRef = React.createRef<ScrollView>();
-  private scrollToTop = () => {
-    if (this.WalletLayoutRef.current) {
-      this.WalletLayoutRef.current.scrollTo({ x: 0, y: 0, animated: false });
-    }
-  };
+  private dynamicSubHeader() {
+    return this.props.hasDynamicSubHeader ? (
+      <View style={[styles.whiteBg, styles.flex1]}>
+        <View spacer={true} />
+        <View style={styles.subHeaderContent}>
+          <H5 style={styles.brandDarkGray}>
+            {I18n.t("wallet.latestTransactions")}
+          </H5>
+          <Text>{I18n.t("wallet.total")}</Text>
+        </View>
+        <View spacer={true} />
+      </View>
+    ) : (
+      []
+    );
+  }
 
   public render(): React.ReactNode {
-    return (
-      <Container>
-        <AppHeader style={styles.darkGrayBg} noLeft={!this.props.allowGoBack}>
-          {this.props.allowGoBack && (
-            <Left>
-              <GoBackButton white={true} style={styles.noalias} />
-            </Left>
-          )}
-          <Body>
-            <PagoPALogo />
-          </Body>
-          <Right>
-            <InstabugButtons color={variables.colorWhite} />
-          </Right>
-        </AppHeader>
+    const {
+      title,
+      allowGoBack,
+      topContent,
+      hideHeader,
+      footerContent,
+      contentStyle
+    } = this.props;
 
-        <ScrollView
-          bounces={false}
-          style={
-            this.props.contentStyle ? this.props.contentStyle : styles.whiteBg
+    return (
+      <TopScreenComponent
+        goBack={allowGoBack}
+        title={title}
+        dark={true}
+        headerBody={<PagoPALogo />}
+      >
+        <AnimatedScreenContent
+          hideHeader={hideHeader}
+          title={title ? title : I18n.t("wallet.wallet")} // riverifica su altri screen
+          icon={require("../../../img/wallet/bank.png")} // verifica se utilizzato
+          dark={true}
+          dynamicSubHeader={this.dynamicSubHeader()}
```

Comment:
You are using a AnimatedScreenContent also if hasDynamicSubHeader is false. Isn't better to check hasDynamicSubHeader and if true render an AnimatedScreenContent if false render a normal ScreenContent.

Reply:
done! I created a unique function which provides del screen content to both the animated and not animated components to avoid redundancy.